scio-test/src/it/scala/com.spotify.scio.values/DistCacheTest.scala

```diff
@@ -0,0 +1,21 @@
+package com.spotify.scio.values
+
+import com.spotify.scio.testing.ITPipelineSpec
+
+class DistCacheTest extends ITPipelineSpec {
+
+  "GCS DistCache" should "work" in {
+    val distCacheUri = "gs://data-integration-test-us/scio/name-distcache"
+
+    runWithDistCache(Seq("name1", "name2"), distCacheUri) { sc =>
```

Comment:
`runWithDistCache` treats `data` as strings but the reading logic is here. This could be a potential cause of confusion.
I'd change the 2nd param from `(fn: ScioContext => T2)` to `(fn: (ScioContext, DistCache[T1]) => T2)` and move reading logic to `runWithDistCache`. That way you can get rid of `distCacheUri` too and use a temp URI inside `runWithDistCache`.